emchat-server-nodejs/easemob.js

```diff
@@ -1,73 +1,1010 @@
+		target:json.target,
+		msg:{type:'txt',msg:json.content},
+		from:json.from,
+	};
+	json.ext && (data.ext=json.ext);
+	http_request({
+		data:data,
+		path:'/messages',
+		method:'POST',
+		callback:function(data){
+			console.log(data);
+			json.callback && json.callback(data);
+		}	
+	});
+}
+
+//发送图片消息
+/*
+	json.type:
+	json.target:
+	json.url:
+	json.filename:
+	json.secret:
+	json.from:
+	json.ext:
+	json.callback:
+	
+*/
+function sendImage(json){
+	var json = json || {};
+	var data={
+		target_type:json.type,
+		target:json.target,
+		msg:{type:'img',url:json.url,filename:json.filename,secret:json.secret,size:{width:480,height:720}},
+		from:json.from,
+	};
+	json.ext && (data.ext=json.ext);
+	http_request({
+		data:data,
+		path:'/messages',
+		method:'POST',
+		callback:function(data){
+			console.log(data);
+			json.callback && json.callback(data);
+		}	
+	});
+}
+
+//发送语音消息
+/*
+	json.type:
+	json.target:
+	json.url:
+	json.filename:
+	json.length:
+	json.secret:
+	json.from:
+	json.ext:
+	json.callback:
+	
+*/
+function sendAudio(json){
+	var json = json || {};
+	var data={
+		target_type:json.type,
+		target:json.target,
+		msg:{type:'audio',url:json.url,filename:json.filename,length:json.length,secret:json.secret},
+		from:json.from,
+	};
+	json.ext && (data.ext=json.ext);
+	http_request({
+		data:data,
+		path:'/messages',
+		method:'POST',
+		callback:function(data){
+			console.log(data);
+			json.callback && json.callback(data);
+		}	
+	});
+}
+
+//发送视频消息
+/*
+	json.type:
+	json.target:
+	json.url:
+	json.filename:
+	json.thumb:
+	json.length:
+	json.file_length:
+	json.thumb_secret:
+	json.secret:
+	json.from:
+	json.ext:
+	json.callback:
+	
+*/
+function sendVedio(json){
+	var json = json || {};
+	var data={
+		target_type:json.type,
+		target:json.target,
+		msg:{type:'video',url:json.url,filename:json.filename,thumb:json.thumb,length:json.length,file_length:json.file_length,thumb_secret:json.thumb_secret,secret:json.secret},
+		from:json.from,
+	};
+	json.ext && (data.ext=json.ext);
+	http_request({
+		data:data,
+		path:'/messages',
+		method:'POST',
+		callback:function(data){
+			console.log(data);
+			json.callback && json.callback(data);
+		}	
+	});
+}
+
+//发送透传消息
+/*
+	json.type:
+	json.target:
+	json.action:
+	json.from:
+	json.ext:
+	json.callback:
+	
+*/
+function sendCmd(json){
+	var json = json || {};
+	var data={
+		target_type:json.type,
+		target:json.target,
+		msg:{type:'cmd',action:json.action},
+		from:json.from,
+	};
+	json.ext && (data.ext=json.ext);
+	http_request({
+		data:data,
+		path:'/messages',
+		method:'POST',
+		callback:function(data){
+			console.log(data);
+			json.callback && json.callback(data);
+		}	
+	});
+}
+
+//---------------------------------------------------------------群组操作
+
+//获取所有群组
+function getGroups(callback){
+	http_request({
+		path:'/chatgroups',
+		method:'GET',
+		callback:function(data){
+			console.log(data);
+			callback && callback(data);	
+		}	
+	});
+}
```

Comment:
For the api to fetch all entities will be duprecated, fetch chatgroups with this "/org/app/chatgroups?limit=XX"